gstd/src/prelude.rs

```diff
@@ -19,25 +19,81 @@
 //! The `gstd` default prelude. Re-imports default `std` modules and traits.
 //! `std` can be safely replaced to `gstd` in the Rust programs.
 
-pub use core::prelude::v1::*;
+// Reexports from Rust's libraries
 
-// Public module re-exports
-pub use alloc::{borrow, boxed, collections, fmt, format, rc, slice, string, vec};
+pub use core::prelude::rust_2021::*;
+
+pub use ::alloc::alloc;
+pub use ::alloc::borrow;
+pub use ::alloc::boxed;
+pub use core::any;
+pub use core::array;
+pub use core::ascii;
+pub use core::cell;
+pub use core::char;
+pub use core::clone;
+pub use core::cmp;
+/// Collection types.
+///
+/// See [`alloc::collections`] & [`hashbrown`].
+///
+/// [`alloc::collections`]: ::alloc::collections
+pub mod collections {
+    pub use ::alloc::collections::*;
+    pub use ::hashbrown::{hash_map, hash_set, HashMap, HashSet};
+
+    /// Reexports from [`hashbrown`].
+    pub mod hashbrown {
+        pub use hashbrown::{Equivalent, TryReserveError};
+    }
+}
+pub use core::convert;
+pub use core::default;
+/// Utilities related to FFI bindings.
+///
+/// See [`alloc::ffi`] & [`core::ffi`].
+///
+/// [`alloc::ffi`]: ::alloc::ffi
+pub mod ffi {
+    pub use ::alloc::ffi::*;
+    pub use core::ffi::*;
+}
+pub use ::alloc::fmt;
+pub use ::alloc::rc;
+pub use ::alloc::str;
+pub use ::alloc::string;
+pub use ::alloc::vec;
+pub use core::future;
+pub use core::hash;
+pub use core::hint;
+pub use core::iter;
```

Comment:
What the reason to look like std prelude? How this can help smart contracts developers? Also you need in this case to sort it by hands

Reply:
Adding additional re-exporting is good. My question is only about fmt